app/routes.py

```diff
@@ -1,2 +1,290 @@
-from flask import Blueprint
+from flask import Blueprint, request, make_response, jsonify
+from app.models.task import Task
+from app.models.goal import Goal
+from app import db
+from datetime import datetime
+import os
+import requests
 
+
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+# I am CHUFFED with this one
+def post_to_slack(text, request_method):
+    """
+    Posts a message to Slack channel
+    text: string to be posted
+    request_method: type of request, e.g. requests.patch
+    """
+    headers = {"Authorization": os.environ.get("SLACK_KEY")}
+    data = {
+        "channel": "C021GPYFGKT",
+        "text": text
+    }
+
+    request_method('https://slack.com/api/chat.postMessage',
+    headers=headers, data=data)
+
+def validate_datetime(user_input):
+    """
+    Validates completed_at input to ensure it is either None
+    or in proper datetime format
+    """
+    if not user_input["completed_at"]: # could/should this be a try/except clause?
+        if not isinstance(user_input["completed_at"], datetime)\
+        and user_input["completed_at"] != None:
+            return make_response("completed_at must be in correct format", 401)
+
+@tasks_bp.route("", methods=["GET"])
+def get_tasks():
+    """
+    Gets all tasks, with option of querying by ascending or descending order
+    """
+    sort_query = request.args.get("sort")
+    if sort_query == "asc":
+        tasks = Task.query.order_by(Task.title)
+    elif sort_query == "desc":
+        tasks = Task.query.order_by(Task.title.desc())
+    else:
+        tasks = Task.query.all()
+
+    tasks_response = []
+    for task in tasks:
+        tasks_response.append(task.to_json())
+    return jsonify(tasks_response)
```

Comment:
Just that this would be a great spot for a list comprehension.

```suggestion
    return [task.to_json() for task in tasks]
```